### PR TITLE
fix!: Make user name nullable in user info.

### PR DESCRIPTION
**Pull request by Isakdl**
# Changes

Makes `userName` nullable in `UserInfo`.

Closes: https://github.com/serverpod/serverpod/issues/975

## Pre-launch Checklist

- [x] I read the [Contribute](https://docs.serverpod.dev/contribute) page and followed the process outlined there for submitting PRs.
- [x] This update contains only one single feature or bug fix and nothing else. (If you are submitting multiple fixes, please make multiple PRs.)
- [x] I read and followed the [Dart Style Guide](https://dart.dev/guides/language/effective-dart/style) and formatted the code with [dart format](https://dart.dev/tools/dart-format).
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`), and made sure that the documentation follows the same style as other Serverpod documentation. I checked spelling and grammar.
- [ ] I added new tests to check the change I am making.
- [x] All existing and new tests are passing.
- [x] Any breaking changes are documented below.

If you need help, consider asking for advice on the [discussion board](https://github.com/serverpod/serverpod/discussions).

## Breaking changes

userName is nullable

### CODE REVIEW

File: tests/serverpod_test_server/migrations/migration_registry.txt [hunk 1]
```diff
@@ -5,3 +5,4 @@
 ### the conflict by removing and recreating the conflicting migration.
 
 20240125141253049
+20240520103319212
```
Comment (SandPod) [inline tests/serverpod_test_server/migrations/migration_registry.txt:8]: Discussion: Would be good if we have a single migration for each major version update of Serverpod maybe?

**Review by SandPod — APPROVED** [review]
Changes look good!

Two general questions:
1. What is the default behavior for the user identifier for the providers? Should we change approach there as well?
2. UI changes are generally quite hard to review. Have you tested that these changes render correctly? A way to both certify that the changes have been tested and make it a bit more clear what has changes is if we would include printscreens of the changes in the PR.

File: tests/serverpod_test_server/migrations/migration_registry.txt (hunk 1 above)
Comment (Isakdl) [reply tests/serverpod_test_server/migrations/migration_registry.txt:8]: Probably yes.

**Comment by Isakdl** [comment]
User identifier has not default, it is up to the provider to set one. I think for email/password it is the email, for apple it is the apple id, firebase the firebase uid, google I think is email.

On the UI I should have added a screenshot indeed. Would be cool if we had widgetbook for the screens / widgets. Then we could host a review page for each PR.

**Pull request merged**